Doppler.ContactPolicies.Business.Logic/Services/ContactPoliciesService.cs

```diff
@@ -21,5 +22,36 @@ public async Task<ContactPoliciesSettingsDto> GetContactPoliciesSettingsAsync(st
                 (await _contactPoliciesSettingsRepository.GetContactPoliciesSettingsAsync(accountName)).ToDto();
             return contactPoliciesSettings;
         }
+
+        public async Task<bool> UpdateContactPoliciesSettingsAsync(string accountName, ContactPoliciesSettingsDto contactPoliciesSettings)
+        {
+            var existingUserContactPolicies = await _contactPoliciesSettingsRepository.GetBasicContactPoliciesSettingsAsync(accountName);
+
+            // User doesnt exist
+            if (existingUserContactPolicies is null)
+                return false;
+
+            // User doesnt have permissions
+            if (existingUserContactPolicies.IdUser is null)
+                throw new Exception($"This action is not allowed for the user with Account {accountName}.");
+
+            var contactPoliciesToInsert = CreateUserContactPoliciesToUpdate(contactPoliciesSettings, existingUserContactPolicies);
+            var isSuccessfullyInserted = await _contactPoliciesSettingsRepository.UpdateContactPoliciesSettingsAsync(contactPoliciesToInsert);
+
+            return isSuccessfullyInserted;
```

Comment:
What do you think about this?
```suggestion
            var contactPoliciesToUpdate = CreateUserContactPoliciesToUpdate(contactPoliciesSettings, existingUserContactPolicies);
            var isSuccessfullyUpdated = await _contactPoliciesSettingsRepository.UpdateContactPoliciesSettingsAsync(contactPoliciesToInsert);

            return isSuccessfullyUpdated;
```